src/main/java/frc/robot/commands/feeder/FeederGamepieceUntilLeave.java

```diff
@@ -28,12 +30,14 @@ public void initialize() {
     }
     @Override
     public boolean isFinished() {
-        if (Timer.getFPGATimestamp() - time > 0.5) {
-            return feeder.pieceNotSeen() || Timer.getFPGATimestamp() - time > 5;
+        if (feeder.pieceNotSeen()) {
```

Comment:
The logic is wrong here. 
This command should run the feeder for 0.5 seconds and only then it should either stop if it doesn't see a game piece or timeout.

Reply:
Done

Reply:
if it times out, you will get out in the :
return feeder.pieceNotSeen() || Timer.getFPGATimestamp() - time > 5;
and you're not getting to the else to increment the timer.

Reply:
Fixed